src/sseq.rs

```diff
@@ -142,39 +144,94 @@ impl<const N: usize> SSeqGen<N> {
 
 #[derive(Copy, Clone)]
 pub enum InsertionIterOpt {
-    // Insert N bases
+    // Insert N bases in addition to ACGT
     IncludeNBase,
-    // Do not insert N bases
+    // Do not insert N bases. Only insert ACGT
     ExcludeNBase,
 }
 
+/// Helper struct used to represent a subset of ACGTN characters
+#[derive(Clone, Copy)]
+
+pub struct SSeqChars(SSeqGen<5>);
+
+impl SSeqChars {
+    pub fn new(chars: &[u8]) -> Self {
+        SSeqChars(SSeqGen::from_iter(chars.iter().unique()))
+    }
+    pub fn actg() -> Self {
+        SSeqChars(SSeqGen::from_bytes_unchecked(b"ACGT"))
+    }
+    pub fn actgn() -> Self {
+        SSeqChars(SSeqGen::from_bytes_unchecked(b"ACGTN"))
+    }
+    pub fn n() -> Self {
+        SSeqChars(SSeqGen::from_bytes_unchecked(b"N"))
+    }
+    pub fn none() -> Self {
+        SSeqChars(SSeqGen::new())
+    }
+    pub fn contains(&self, char: &u8) -> bool {
+        self.0.as_bytes().contains(char)
+    }
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
 #[derive(Copy, Clone)]
 pub enum HammingIterOpt {
+    // Mutate the N base. The allowed mutations at each base is ACGT
     MutateNBase,
+    // Don't mutate the N base. The allowed mutations for other bases are ACGT
     SkipNBase,
+    Custom {
+        // Characters to skip.
+        skip_chars: SSeqChars,
+        // Allowed mutations for each base.
+        mutation_chars: SSeqChars,
+    },
+}
+
+impl HammingIterOpt {
+    fn skip_chars(self) -> SSeqChars {
+        match self {
+            HammingIterOpt::SkipNBase => SSeqChars::n(),
+            HammingIterOpt::MutateNBase => SSeqChars::none(),
+            HammingIterOpt::Custom { skip_chars, .. } => skip_chars,
+        }
+    }
+    fn mutation_chars(self) -> SSeqChars {
+        match self {
+            HammingIterOpt::SkipNBase | HammingIterOpt::MutateNBase => SSeqChars::actg(),
+            HammingIterOpt::Custom { mutation_chars, .. } => mutation_chars,
+        }
+    }
 }
 
 /// An iterator over sequences which are one hamming distance away
 /// from an `SSeq`. `SSeq` is guaranteed to contain "ACGTN" alphabets.
 /// Positions containing "N" or "n" are mutated or skipped
 /// depending on the `HammingIterOpt`
 pub struct SSeqOneHammingIter<const N: usize> {
-    source: SSeqGen<N>, // Original SSeq from which we need to generate values
-    position: usize,    // Index into SSeq where last base was mutated
-    chars_index: usize, // The last base which was used
-    skip_n: bool,       // Whether to skip N bases or mutate them
+    source: SSeqGen<N>,    // Original SSeq from which we need to generate values
+    position: usize,       // Index into SSeq where last base was mutated
+    chars: SSeqChars,      // Characters used for mutation
+    chars_index: usize,    // The last base which was used
+    skip_chars: SSeqChars, // Characters to skip
```

Comment:
Even for private struct members, use `///` docstrings; that ensures they appear in intellisense when you hover over the struct field, which can be very helpful when working with unfamiliar code.

Reply:
Oh, good point!